scripts/ci/push-registry.py

```diff
@@ -0,0 +1,292 @@
+    "opentelekomcloud": "opentelekomcloud",
+    "oun": "oun",
+    "outscale": "outscale",
+    "paloaltonetworks": "paloaltonetworks",
+    "paloaltonetworks": "paloaltonetworks",
+    "pan-net": "pan-net",
+    "paultyng": "paultyng",
+    "pgEdge": "pgedge",
+    "philips-software": "philips-software",
+    "pinecone-io": "pinecone",
+    "planetscale": "planetscale",
+    "port-labs": "port_labs",
+    "prefecthq": "prefecthq",
+    "propelauth": "propelauth",
+    "pulumiverse - Marcel Arns": "pulumiverse",
+    "pulumiverse": "pulumiverse",
+    "qdrant": "qdrant",
+    "rancher": "rancher",
+    "redpanda-data": "redpanda_data",
+    "rollbar": "rollbar",
+    "selectel": "selectel",
+    "spectrocloud": "spectrocloud",
+    "splightplatform": "splightplatform",
+    "supabase": "supabase",
+    "sysdiglabs": "sysdiglabs",
+    "temporalio": "temporalio",
+    "tencentcloudstack": "tencentcloudstack",
+    "terraform-lxd": "terraform_lxd",
+    "terraform-routeros": "terraform-routeros",
+    "timescale": "timescale",
+    "ucloud": "ucloud",
+    "vantage-sh": "vantage-sh",
+    "vk-cs": "vk-cs",
+    "vmware": "vmware",
+    "vmware": "vmware",
+    "vmware": "vmware",
+    "vmware": "vmware",
+    "vmware": "vmware",
+    "zenduty": "zenduty",
+}
+
+has_failed = False
+
+for filename in os.listdir(yaml_db_path):
+    if not filename.endswith('.yaml'):
+        continue
+    if args.target and args.target != filename.removesuffix(".yaml"):
+        continue
+
+    filepath = os.path.join(yaml_db_path, filename)
+    with open(filepath) as file:
+        data = yaml.safe_load(file)
+
+    version = data.get('version')
+
+    schema_file_path = data.get('schema_file_path')
+    schema_url = data.get(
+        'schema_file_url',
+        f"https://raw.githubusercontent.com/{data.get('repo_url').removeprefix("https://github.com/")}/refs/tags/{version}/{schema_file_path}",
+    )
+    
+    source = 'pulumi' if 'registry.opentofu.org' not in schema_url else "opentofu"
+    name = data.get('name')
+    publisher_display = data.get('publisher')
+
+
+    if publisher_display == "DEPRECATED":
+        continue
+    elif publisher_display not in publishers:
+        raise Exception(f'Missing publisher entry for "{publisher_display}"')
+    publisher = publishers[publisher_display]
+
+
+    package_version_name = f"{source}/{publisher}/{name}@{version.removeprefix("v")}"
+    info(f"Preparing command for {package_version_name}")
+
+    # We skip azure-native-v1, since it will (1) never be updated again and (2) isn't
+    # actually a package, it's an alias to the azure native package.
+    #
+    # If azure-native-v2 ships (with azure-native now at v3), we will want to exclude
+    # azure-native-v2 also.
+    if name.startswith("azure-native") and name != "azure-native":
+        continue
+
+
+    api_url = f"https://api.pulumi.com/api/preview/registry/packages/{source}/{publisher}/{name}/versions/{version.removeprefix("v")}"
```

Comment:
Nit: It would be great to make the base endpoint configurable. E.g. for testing against a devstack or review stack